### ISSUE TITLE
[Bug]: Some packages use sudo when installing dependencies in $PREFIX

### ISSUE THREAD

**Issue by Yisus7u7** [termux/termux-packages#16467]
### Problem description

When installing some dependency (for example, in Perl using cpan or in Ruby using gem), when using the global installation method, if the tsu package is installed, the sudo command will be executed during the installation.

It also happens with npm from nodejs, but in nodejs you can manually change the global installation path.

### What steps will reproduce the bug?

Just install stuff with npm, gem or cpan with the tsu package installed, on some occasions you'll see sudo commands run during installation.

### What is the expected behavior?

When installing a module or package with any of these managers, commands must not be executed with root permissions

A possible solution to this that seems correct to me is to change the name of the "sudo" command to something else to avoid problems like these.

### System information

termux-info:

```
Termux Variables:
TERMUX_APP_PACKAGE_MANAGER=pacman               TERMUX_APP__AM_SOCKET_SERVER_ENABLED=true
TERMUX_APP__APK_PATH=/data/app/~~Oi1xpYZdGkwDyV9NldS_Rg==/com.termux-ySrvlhqzlOlaFFhT2QqG2w==/base.apk
TERMUX_APP__APK_RELEASE=GITHUB                  TERMUX_APP__FILES_DIR=/data/user/0/com.termux/files
TERMUX_APP__IS_DEBUGGABLE_BUILD=true            TERMUX_APP__IS_INSTALLED_ON_EXTERNAL_STORAGE=false
TERMUX_APP__PACKAGE_MANAGER=apt                 TERMUX_APP__PACKAGE_NAME=com.termux
TERMUX_APP__PACKAGE_VARIANT=apt-android-7
TERMUX_APP__PID=18365                           TERMUX_APP__SE_FILE_CONTEXT=u:object_r:app_data_file:s0:c177,c256,c512,c768                     TERMUX_APP__SE_INFO=default:targetSdkVersion=28:complete
TERMUX_APP__SE_PROCESS_CONTEXT=u:r:untrusted_app_27:s0:c177,c256,c512,c768
TERMUX_APP__TARGET_SDK=28
TERMUX_APP__UID=10177
TERMUX_APP__USER_ID=0
TERMUX_APP__VERSION_CODE=118
TERMUX_APP__VERSION_NAME=0.118.0+93eafff
TERMUX_MAIN_PACKAGE_FORMAT=pacman
TERMUX_VERSION=0.118.0+93eafff
Packages CPU architecture:
aarch64
Subscribed repositories:
# /data/data/com.termux/files/usr/etc/pacman.conf
[main]
Usage = All
Server = https://s3.amazonaws.com/termux-pacman.us/main/aarch64
[x11]
Usage = All
Server = https://s3.amazonaws.com/termux-pacman.us/x11/aarch64
[root]
Usage = All
Server = https://s3.amazonaws.com/termux-pacman.us/root/aarch64
[tur]
Usage = All
Server = https://s3.amazonaws.com/termux-pacman.us/tur/aarch64
[tur-continuous]
Usage = All
Server = https://s3.amazonaws.com/termux-pacman.us/tur-continuous/aarch64
[tur-multilib]
Usage = All
Server = https://s3.amazonaws.com/termux-pacman.us/tur-multilib/aarch64
Updatable packages:
All packages up to date
termux-tools version:
1.38.1
Android version:
13
Kernel build information:
Linux localhost 4.4.302-lineage-gd54468d17934 #1 SMP PREEMPT Tue Apr 25 07:27:59 UTC 2023 aarch64 Android
Device manufacturer:
motorola
Device model:
moto g(6) plus
LD Variables:
LD_LIBRARY_PATH=
LD_PRELOAD=/data/data/com.termux/files/usr/lib/libtermux-exec.so
```

**Comment by thunder-coding** [comment]
> When installing some dependency (for example, in Perl using cpan or in Ruby using gem), when using the global installation method, if the tsu package is installed, the sudo command will be executed during the installation.

Are you sure that the sudo command is not coming from post install scripts? If so there's nothing we can do from our side, aside from asking the maintainer of the package to add an option to not use sudo maybe from environment variable or something.

**Comment by Yisus7u7** [comment]
> > When installing some dependency (for example, in Perl using cpan or in Ruby using gem), when using the global installation method, if the tsu package is installed, the sudo command will be executed during the installation.
> 
> Are you sure that the sudo command is not coming from post install scripts? If so there's nothing we can do from our side, aside from asking the maintainer of the package to add an option to not use sudo maybe from environment variable or something.

An easily reproducible example.

Install some package with cpan, when it asks for the installation method, (sudo, local:lib) I used the sudo method so that the module is installed in $PREFIX.

If you have tsu installed, during the installation you will see it run sudo several times writing files with root permissions to $PREFIX.

**Comment by xtkoba** [comment]
As for gem, IIRC the "auto-sudo" feature has been removed. Which version are you using?

```bash
bundler --version
```

See also https://github.com/rubygems/rubygems/pull/5888.

**Comment by xtkoba** [comment]
Issues of individual `sudo` consumers. `tsu` should not need to be modified. Please file a separate issue for each package. Thanks.

Note that the examples in the description are no longer issues. `cpan` in `perl` has already been fixed (#16475). The issue with `gem` in `ruby` is not reproducible with the latest version in our repo.
